components/org.wso2.micro.integrator.initializer/src/main/java/org/wso2/micro/integrator/initializer/deployment/synapse/deployer/FileRegistryResourceDeployer.java

```diff
@@ -179,6 +205,29 @@ private void writeArtifactToRegistry(RegistryConfig registryConfig){
         }
     }
 
+    /**
+     * Remove all registry contents (resources) of the given artifact from the registry.
+     *
+     * @param registryConfig - Artifact instance
+     */
+    private void removeArtifactFromRegistry(RegistryConfig registryConfig){
+
+        // get resources
+        List<RegistryConfig.Resourse> resources = registryConfig.getResources();
+        for (RegistryConfig.Resourse resource : resources) {
+            String filePath = registryConfig.getExtractedPath() + File.separator + AppDeployerConstants.RESOURCES_DIR
+                    + File.separator + resource.getFileName();
+            // check whether the file exists
+            File file = new File(filePath);
+            if (!file.exists()) {
+                log.error("Specified file to be removed as a resource is " + "not found at : " + filePath);
```

Comment:
when clustered , another node would have deleted it already. should we simply ignore this rather than printing the error msg or debug ?

Reply:
+1.